source/byte_buf.c

```diff
@@ -405,28 +413,39 @@ bool aws_byte_cursor_eq_c_str_ignore_case(const struct aws_byte_cursor *cursor,
 }
 
 int aws_byte_buf_append(struct aws_byte_buf *to, const struct aws_byte_cursor *from) {
-    assert(from->ptr);
-    assert(to->buffer);
+    AWS_PRECONDITION(aws_byte_buf_is_valid(to));
+    AWS_PRECONDITION(aws_byte_cursor_is_valid(from));
+    if (!from->ptr || !to->buffer) {
```

Comment:
do we need this on top of the existing precondtions

Reply:
Yes, because we consider empty `buf` or `cursor` as valid. However, `aws_byte_buf_append` does not work on empty structures. Of course, we could also update the implementation to consider them. What would you suggest?